### PR TITLE
Data100 pivot_table element packaged

### PR DESCRIPTION
**Pull request by ownssi**
Hi! This is pull request for packaging. Please let me know if there is anything gone wrong on packaging.

### CODE REVIEW

File: elements/pl-pivot-table/pl-pivot-table-double.mustache [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+        <div class="container index-box">
+            <h2>Choose answer index here</h2>
+            <div class="row">
+                {{#indice_set}}
+                <div class="col-md-1"{{{calibration}}}>
+                    <div class="index-set" id="{{uuid}}" data-order="{{order_index}}">
+                        {{#index}}
+                        <div class="row">
+                            <div class="col-md-12 index-cell" style="height: 30px;">{{{inner_html}}}</div>
```
Comment (armandofox) [inline elements/pl-pivot-table/pl-pivot-table-double.mustache:30]: please do not use inline styles at all.  and definitely do not use fixed heights like"30px", which breaks if the user resizes their font or changes their zoom level.  use the `-py` and `-my` and similar classes in bootstrap if you need to adjust vertical spacing or height.

File: elements/pl-pivot-table/pl-pivot-table-double.mustache [hunk 2]
```diff
@@ -0,0 +1,106 @@
+{{#question}}
+        <script>
```
Comment (armandofox) [inline elements/pl-pivot-table/pl-pivot-table-double.mustache:2]: javascript should not be inline.  any separate .js files need to be in the element's directory and should be loaded via `script src=`.

File: elements/pl-pivot-table/pl-pivot-table-double.mustache [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+                        </div>
+                        {{/is_ellipsis}}
+                    </div>
+                </div>
+                {{/row_set}}
+            </div>
+        </div>
+
+        <div class="container drop-zone">
+            <h2>Drop your answer here</h2>
```
Comment (armandofox) [inline elements/pl-pivot-table/pl-pivot-table-double.mustache:72]: there is a huge amount of stuff in this template that is fixed and not customizable.  it seems to me that at a minimum, things like the prompts (names of sections such as "Drop your answer here") should be customizable.  many elements allow this; look at `pl-order-blocks` as an example.

File: elements/pl-pivot-table/pl-pivot-table-single.mustache [hunk 4]
```diff
@@ -0,0 +1,132 @@
+{{#question}}
+        <script>
```
Comment (armandofox) [inline elements/pl-pivot-table/pl-pivot-table-single.mustache:2]: same comment about inline jS

File: elements/pl-pivot-table/pl-pivot-table-single.mustache [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+                {{/column}}
+                {{#is_ellipsis}}
+                <div class="col-md-{{{width}}} col-cell">...</div>
+                {{/is_ellipsis}}
+            </div>
+            {{/column_set}}
+        </div>
+
+        <div class="container index-box">
+            <h2>Choose answer index here</h2>
```
Comment (armandofox) [inline elements/pl-pivot-table/pl-pivot-table-single.mustache:23]: same comment about customizing section names

**Review by armandofox — CHANGES_REQUESTED** [review]
- what exactly is the difference between the two mustache templates -single and -double?  it looks like there is a huge amount of repeated code.  is there no way to DRY this out?
- i made numerous small comments in the template files.
- the README documentation contains a very large number of grammar errors.  please get some assistance to fix these since this is the pubilc documentation that others will use to develop questiosn with the element. it needs to be at least mostly correct and very clear.

File: elements/pl-pivot-table/pl-pivot-table-double.mustache (hunk 2 above)
Comment (ownssi) [reply elements/pl-pivot-table/pl-pivot-table-double.mustache:2]: Is inline-js not allowed at all? I have seen built-in pl-element, pl-order-block is using inline-js. I just wonder whether there is any potential issue from inline-js.

https://github.com/PrairieLearn/PrairieLearn/blob/1e3deb438b2c2b95c6291811afe918f7144a3935/apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.mustache#L16

**Comment by matthewlee626** [comment]
Tried to lint the readme here and add some clarity to the sections here.

File: elements/pl-pivot-table/pl-pivot-table-double.mustache (hunk 1 above)
Comment (ownssi) [reply elements/pl-pivot-table/pl-pivot-table-double.mustache]: This is resolved

File: elements/pl-pivot-table/pl-pivot-table-double.mustache (hunk 2 above)
Comment (ownssi) [reply elements/pl-pivot-table/pl-pivot-table-double.mustache]: This is resolved

File: elements/pl-pivot-table/pl-pivot-table-double.mustache (hunk 3 above)
Comment (ownssi) [reply elements/pl-pivot-table/pl-pivot-table-double.mustache]: This is resolved

File: elements/pl-pivot-table/pl-pivot-table-single.mustache (hunk 4 above)
Comment (ownssi) [reply elements/pl-pivot-table/pl-pivot-table-single.mustache]: This is resolved

File: elements/pl-pivot-table/pl-pivot-table-single.mustache (hunk 5 above)
Comment (ownssi) [reply elements/pl-pivot-table/pl-pivot-table-single.mustache]: This is resolved

**Review by armandofox — CHANGES_REQUESTED** [review]
this looks good, a few minor changes:
- please remove all .DS_Store files, those are created by macOS and should never be versioned. in general, never use `git add .` to add files - add them one at a time to avoid these problems.  i will also modify the repo's .gitignore to avoid adding these in the future
- i think the question names should use either all hyphens or all underscores, ie `pivot-table-question1` or `pivot_table_question1`, not a mixture.  either is ok.  when changing this please adjust `infoAssessment.json` as needed too.
-

**Comment by ownssi** [comment]
Replaced question names' underscores by hyphens, and checked all possible folders where .DS_Store can be.

**Pull request merged**